Makefile

```diff
@@ -1,9 +1,3 @@
-# Define PETSC_DIR to ~/petsc if it's not defined
-ifdef PETSC_DIR
-else
-PETSC_DIR = $(HOME)/petsc
-endif
-
 include ${PETSC_DIR}/lib/petsc/conf/variables
```

Comment:
```
ifndef PETSC_DIR
$(error PETSC_DIR environment variable is not set. Please, set it before continue.)
endif

ifndef PETSC_ARCH
$(error PETSC_ARCH environment variable is not set. Please, set it before continue.)
endif
```
Reference: https://www.gnu.org/software/make/manual/make.html#Make-Control-Functions

Maybe we can add these checks. This trigger an error if we try to run any action and this variables are not defined.

Reply:
I like this, @rafaelmds!

I will add the first error message. But the second one is not necessary because the PERSC_ARCH is imported in this line of the Makefile:
```
include ${PETSC_DIR}/lib/petsc/conf/variables
include ${PETSC_DIR}/lib/petsc/conf/rules
```

Reply:
great!

But I don't think PETSC_ARCH is included:
```sh
$ grep 'PETSC_ARCH' ${PETSC_DIR}/lib/petsc/conf/variables
PETSC_LIB_DIR	     = ${PETSC_DIR}/${PETSC_ARCH}/lib
PETSCCONF_H          = ${PETSC_DIR}/${PETSC_ARCH}/include/petscconf.h
include ${PETSC_DIR}/${PETSC_ARCH}/lib/petsc/conf/petscvariables
```
Some info about, in which docs recommends to export PETSC_DIR and PETSC_ARCH for in-place installs:
https://petsc.org/release/install/multibuild/#environmental-variables-petsc-dir-and-petsc-arch

Reply:
Correct - for the most generality with PETSc it should work with both in-place installs, where you specify both `PETSC_DIR` and `PETSC_ARCH`, and out-of-place (aka "prefix") installs where you only specify `PETSC_DIR` (and should *not* define `PETSC_ARCH` so that part of the paths above is empty).

Reply:
thanks @psanan !